workspaces/diff-engine/src/commands/mod.rs

```diff
@@ -188,7 +190,8 @@ impl AggregateCommand<SpecProjection> for SpecCommand {
         };
 
         let shape_added_event = ShapeEvent::from(ShapeCommand::add_shape(
-          String::from("$string"),
+          Uuid::new_v4().to_hyphenated().to_string(),
```

Comment:
do you want to use the same id generator everywhere?

Reply:
Would make sense, wouldn't it? Especially with a trait generically named `SpecIdGenerator`. I'll get that cleaned up and consistent.

Reply:
I'm going to cover this one in a separate PR, as it involves a restructuring of the command handling (moving all of `SpecProjection` implementation of `AggregationCommand` to the `SpecCommandHandler`). Totally worth doing, but given this PR has already become massive with main goal of killing Scala.